### DISCUSSION TITLE
First Time Upgrading Budibase to Newer Release (Docker Platform)

### DISCUSSION THREAD

**Discussion by chadgriff** [Budibase/budibase#2873] [Help] [answered]
I setup and configured Budibase earlier this year using Docker.  However, I would now like to upgrade to the August release and I find myself struggling to get it done.  I have built a simple app and would like to know a bit more about the process to minimize the chance of losing my work. I have updated Docker to its latest release so I believe I am good on that front.  Just need some guidance on how to get the files from GIT, where to place them, and where to access the appropriate terminal to execute the upgrade commands.  Thanks in advance.

**Comment by mjashanks** [comment]
Hi Chad.

Depending on when you installed Budibase, you might be able to update it via the UI:
![image](https://user-images.githubusercontent.com/3524181/135825168-0b1a7c50-0027-4736-b6e7-d0a33b58aad5.png)

If yo do not see this, or it does not work for you, then you'll need to do it manually.

This will depend on what way you installed Budibase - probably either `docker-compose` or using the CLI (`budi hosting --start`).

In either case, you should have a directory somewhere where you have some Budibase files:
- docker-compose.yaml
- envoy.yaml
- hosting.properties
- .env
First, locate this directory and `cd` into it.

**If using the Budibase CLI**
I use the Budibase CLI, so what I do is:
```bash
sudo budi hosting --update
```

**If using docker compose**
Download these 2 files:
- https://raw.githubusercontent.com/Budibase/budibase/master/hosting/docker-compose.yaml
- https://raw.githubusercontent.com/Budibase/budibase/master/hosting/envoy.yaml

Then, run `docker-compose --env-file hosting.properties up`

These files have changed a bit over time, so don't worry if you encounter an issue - come back here and we'll get you sorted.

**Reply by chadgriff** [reply]
Mjashanks,

Thanks for the directions.  I ran `docker-compose  --env-file hosting.properties up` and immediately got an error stating it couldn't locate a hosting.properties file.  I created an empty hosting.properties file and reran the 'docker-compose --env-file hosting.properties up' command.  It ran and generated the following series of warnings:

![image](https://user-images.githubusercontent.com/15015476/135905216-83473a0b-0abe-44cd-8265-2b8946ce7062.png)

Please advise....

**Reply by PGLange** [reply]
hey @chadgriff ,
try to run 
head to your bbase directory,
cp hosting.properties .env && budi hosting --start
and make sure your enyoy.yaml is existing.

I hope this'll help you,
Philip

**Reply by mjashanks** [reply]
... also, if you have a hidden file called `.env` in that directory, you should just run `docker-compose up`.

If you find that powershell complains about permissions, you might need to run it as administrator first.

**Reply by chadgriff** [reply]
Thanks PGLange and mjashanks.  I have made progress.  I am only getting a single warning now regarding the WATCHTOWER_PORT.
![image](https://user-images.githubusercontent.com/15015476/135917227-a58bd594-e294-4df9-be4b-2672460adfb9.png)

**Reply by mjashanks** [reply]
So, inside your `.env`, add a watchtower line, as per this file.. https://github.com/Budibase/budibase/blob/develop/hosting/hosting.properties

**Reply by chadgriff** [reply]
Still more progress...now I received a permissions error related to the logs file...
![image](https://user-images.githubusercontent.com/15015476/135919305-197dc649-ae46-4dad-969f-0aa0dd8665c2.png)

**Reply by chadgriff** [reply]
I created the \logs folder by hand and now re-running docker-compose up. Things seem to be moving along now.

**Reply by chadgriff** [reply]
Budibase has been upgraded to the latest release but I am getting an error trying to load my client library...

![image](https://user-images.githubusercontent.com/15015476/135921143-8c7db359-5beb-41d3-9b1f-e73dbc0eaa75.png)

**Reply by mjashanks** [reply]
See the little blue refresh arrow in the top-right? Click that - it will update your client library.

![image](https://user-images.githubusercontent.com/3524181/135986991-7a4e2c59-e97f-46df-bd26-071c799957de.png)

**Reply by chadgriff** [reply]
Okay...thanks for sticking with me on this.  I attempted to update the client library and received an error.  Any additional guidance would be appreciated.
![image](https://user-images.githubusercontent.com/15015476/136032789-2711a299-b5fe-4f5a-a85f-602a16b134c7.png)

**Reply by chadgriff** [reply]
Also adding this additional info....
![image](https://user-images.githubusercontent.com/15015476/136036267-58d249f7-3a33-46c3-8d41-49197d47e4d9.png)

**Reply by mjashanks** [reply]
Ok - we did have some breaking changes since this version. Next thing to try - 
1. Export the app
2. Import it to a new app.

You will probably see a few thing that are broken, but won't be too difficult to fix.

**Comment by chadgriff** [comment, marked as answer]
Thanks...i could use the practice in getting familar with the new functionality so I just rebuilt my original app.  All is working good. Thanks so much for your help throughout this process.

**Reply by mjashanks** [reply]
haha good stuff, hopefully didn't take you too long - that's what we're all about!
